packages/core/src/generator/abstract-generator.adapter.ts

```diff
@@ -0,0 +1,134 @@
+import { multiInject } from 'inversify';
+import {
+  GeneratorAdapter,
+  GenerateSectionsOptions,
+} from './generator.adapter.js';
+import {
+  SECTION_GENERATOR_ADAPTER_IDENTIFIER,
+  SectionGeneratorAdapter,
+  SectionOptionsDescriptors,
+  SectionGenerationPayload,
+} from './section-generator.adapter.js';
+import { RepositoryProvider } from '../repository/repository.provider.js';
+import { RendererAdapter } from '../renderer/renderer.adapter.js';
+import { ManifestVersion } from '../version/version.service.js';
+
+/**
+ * Abstract base class for generator adapters to avoid code duplication
+ */
+export abstract class AbstractGeneratorAdapter<TManifest> implements GeneratorAdapter {
+  constructor(
+    @multiInject(SECTION_GENERATOR_ADAPTER_IDENTIFIER)
+    protected readonly sectionGeneratorAdapters: SectionGeneratorAdapter<TManifest>[]
```

Comment:
As it is abstract, do not inject with inversify

```suggestion
    protected readonly sectionGeneratorAdapters: SectionGeneratorAdapter<TManifest>[]
```